package/base-files/files/lib/functions.sh

```diff
@@ -159,6 +159,29 @@ config_foreach() {
 	done
 }
 
+config_find() {
+	local ___function="$1"
+	[ "$#" -ge 1 ] && shift
+	local ___type="$1"
+	[ "$#" -ge 1 ] && shift
+	local section cfgtype
+
+	[ -z "$CONFIG_SECTIONS" ] && return 0
+	for section in ${CONFIG_SECTIONS}; do
+		config_get cfgtype "$section" TYPE
+		[ -n "$___type" ] && [ "x$cfgtype" != "x$___type" ] && continue
```

Comment:
Why the x prefix? I forgot the details. Maybe shellcheck knows more.

Reply:
This is copied from `config_foreach`

Reply:
Actually, almost all code is copied from `config_foreach`

Reply:
I'd check with shellcheck just in case. No sense in introducing potentially faulty code.

Reply:
Checked with shellcheck, it didn't warn about the x, and I've ran the code too.

Reply:
Fair enough